modules/reindex/src/main/java/org/elasticsearch/index/reindex/AbstractAsyncBulkIndexByScrollAction.java

```diff
@@ -86,6 +86,14 @@ protected BulkRequest buildBulk(Iterable<SearchHit> docs) {
         Map<String, Object> scriptCtx = null;
 
         for (SearchHit doc : docs) {
+            if (doc.isSourceEmpty()) {
+                /*
+                 * Either the document didn't store _source or we didn't fetch it for some reason. Since we don't allow the user to
+                 * change the "fields" part of the search request it is unlikely that we got here because we didn't fetch _source.
+                 * Thus the error message assumes that it wasn't stored.
+                 */
+                throw new IllegalArgumentException("[" + doc.index() + "][" + doc.type() + "][" + doc.id() + "] didn't store _source");
```

Comment:
Looking at the impl of isSourceEmpty:

```
    public boolean isSourceEmpty() {
        return source == null;
    }
```
So I think it is badly named and will actually return false if a document is indexed with a json doc that has no fields? It would only return true if the source is not storred at all?